proms-resources/frontend/src/main/frontend/src/proms/PatientIdentification.jsx

```diff
@@ -193,14 +194,22 @@ function PatientIdentification(props) {
     // When the user selects a visit from the visit list, clear the list and
     // send the selected visit back to obtain a token
     setVisitList(null);
+    setVisitListShown(false);
     identify();
   }, [visit,visitList]);
 
+  // After the user has accepted the TOU, if they need to select from a list of visits present said
+  useEffect(() => {
+    if (!visitListShown && touAccepted && visitList && visitList.length > 1 ) {
+      setVisitListShown(true);
+    }
+  }, [visitList, touAccepted]);
+
   // When the visit is successfully obtained and the latest version of Terms of Use accepted, pass it along with the identification data
   // to the parent component
   useEffect(() => {
-    visit && touAccepted && onSuccess && onSuccess(Object.assign({subject: visit}, patientDetails));
-  }, [visit, touAccepted]);
+    visit && touAccepted && patientDetails && onSuccess && onSuccess(Object.assign({subject: visit}, patientDetails));
+  }, [visit, touAccepted, patientDetails]);
```

Comment:
It is recommended to not use objects as a dependency, since strict reference equality is enforced, so this effect may be triggered more than once even if the patient details don't change. Do we only care if it is undefined or not? Would this work?

```suggestion
  }, [visit, touAccepted, !!patientDetails]);
```